models/base_models.py

```diff
@@ -0,0 +1,42 @@
+# MIT License
+#
+# Copyright (c) 2020-2024 Send A Hug
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# The provided Software is separate from the idea behind its website. The Send A Hug
+# website and its underlying design and ideas are owned by Send A Hug group and
+# may not be sold, sub-licensed or distributed in any way. The Software itself may
+# be adapted for any purpose and used freely under the given conditions.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Any, TypeAlias, TypeVar
+
+from sqlalchemy.ext.asyncio import AsyncAttrs
+from sqlalchemy.orm import (
+    DeclarativeBase,
+)
+
+
+class BaseModel(AsyncAttrs, DeclarativeBase):
+    pass
+
+
+HugModelType = TypeVar("HugModelType", bound=BaseModel, covariant=True)
```

Comment:
Two thoughts:
1. `base_models` feels like an inaccurate name considering there's only one model in it. Maybe base/common/something along those lines would be a better fit, what do you think?
2. if we're grouping everything together, which is a good idea, we can probably put the models in db.py here too (unless that's too much of a problem with circular imports)

Reply:
Do you mean CoreSAHModel? That's the only one that looks like a model to me but I might be mistaken

Reply:
have renamed it common - happy to move CoreSAHModel in there

Reply:
Exactly, meant `CoreSAHModel` and the dataclass. That way db.py keeps just the database class, which is what it should've been in the first place

Reply:
cool, moved!